pkgs/build-support/fetchgithub/default.nix

```
@@ -43,17 +43,19 @@ let
     else if fetchzip ? override then fetchzip.override { withUnzip = false; }
     else fetchzip;
   privateAttrs = lib.optionalAttrs private {
-    netrcPhase = ''
-      if [ -z "''$${varBase}USERNAME" -o -z "''$${varBase}PASSWORD" ]; then
-        echo "Error: Private fetchFromGitHub requires the nix building process (nix-daemon in multi user mode) to have the ${varBase}USERNAME and ${varBase}PASSWORD env vars set." >&2
-        exit 1
-      fi
-      cat > netrc <<EOF
-      machine ${githubBase}
-              login ''$${varBase}USERNAME
-              password ''$${varBase}PASSWORD
-      EOF
-    '';
+    netrcPhase =
+      let machineName = if githubBase == "github.com" then "api.github.com" else githubBase;
+      in ''
+        if [ -z "''$${varBase}USERNAME" -o -z "''$${varBase}PASSWORD" ]; then
+          echo "Error: Private fetchFromGitHub requires the nix building process (nix-daemon in multi user mode) to have the ${varBase}USERNAME and ${varBase}PASSWORD env vars set." >&2
+          exit 1
+        fi
+        cat > netrc <<EOF
+        machine ${machineName}
+                login ''$${varBase}USERNAME
+                password ''$${varBase}PASSWORD
```

```suggestion
      in ''
        if [ -z "''$${varBase}USERNAME" -o -z "''$${varBase}PAT" ]; then
          echo "Error: Private fetchFromGitHub requires the nix building process (nix-daemon in multi user mode) to have the ${varBase}USERNAME and ${varBase}PAT env vars set." >&2
          exit 1
        fi
        cat > netrc <<EOF
        machine ${machineName}
                login ''$${varBase}USERNAME
                password ''$${varBase}PASSWORD
```

Just a Suggestion to varBase Password to PATs (Personal Access Token) github api password is deprecated as stated by github
> Beginning August 13, 2021, we will no longer accept account passwords when authenticating Git operations on GitHub.com.

[see the follow blog post for more infomation](https://github.blog/security/application-security/token-authentication-requirements-for-git-operations/)

_**You don't have to take this seriously it's just a suggestion**_

---

I'd rather not: that change would break existing configurations, as `...PASSWORD` has been the variable name for the access token for a long time. And it's still accurate from a protocol perspective: the value gets put in the password part of the HTTP request. The blog post is just that GitHub no longer accepts a GitHub password in the password field.